tests/lib/rules/jsx-sort-props.js

```diff
@@ -494,6 +608,214 @@ ruleTester.run('jsx-sort-props', rule, {
       output: '<App z onBar />;',
       options: reservedFirstAndCallbacksLastArgs,
       errors: [expectedCallbackError],
+    // multiline first
+    },
+    {
+      code: `
+        <App
+          a
+          b={{
+            bB: 1,
+          }}
+        />
+      `,
+      options: multilineFirstArgs,
+      errors: [expectedMultilineFirstError],
+      output: `
+        <App
+          b={{
+            bB: 1,
+          }}
+          a
+        />
+      `,
+    },
+    {
+      code: `
+        <App
+          a={1}
+          b={{
+            bB: 1,
+          }}
+          c
+        />
+      `,
+      options: multilineAndShorthandFirstArgs,
+      errors: [expectedMultilineFirstError, expectedShorthandFirstError],
+      output: `
+        <App
+          c
+          b={{
+            bB: 1,
+          }}
+          a={1}
+        />
+      `,
+    },
+    // multiline last
+    {
+      code: `
+        <App
+          a={{
+            aA: 1,
+          }}
+          b
+        />
+      `,
+      options: multilineLastArgs,
+      errors: [expectedMultilineLastError],
+      output: `
+        <App
+          b
+          a={{
+            aA: 1,
+          }}
+        />
+      `,
+    },
+    {
+      code: `
+        <App
+          a={{
+            aA: 1,
+          }}
+          b
+          inline={1}
```

Comment:
We have a limitation with the current implementation here, whenever conflicting groups are in order between themselves, but in disorder with other props, only the group with higher hierarchy will be reported.

For example here we enabled `{ multiline: 'last', shorthandLast: true }`, but only the shorthand prop is being reported as an error, since multiline thinks that it's in order.

I feel like there's no easy way to fix this unless we change the way in which `jsx-sort-props` checks and order the props, so that it check props by groups instead of the direct adjacent neighbors of each prop. I wouldn't mind working on that, but I feel like that's out of the scope of this PR...

Reply:
I agree; that sounds like a great improvement, but belongs in another PR.